Jenkinsfile

```diff
@@ -298,6 +381,7 @@ pipeline {
                                 echo "PRECOMPILED_HEADERS=false" >> cmdstan/make/local
                                 cd cmdstan; make clean-all; git show HEAD --stat; cd ..
                                 CXX="${CXX}" ./compare-compilers.sh "--tests-file all.tests --num-samples=10" "\$(readlink -f ../bin/stanc)"
+                                CXX="${CXX}" ./compare-optimizer.sh "--tests-file all.tests --num-samples=10" "--O1"
```

Comment:
@WardBrian Do we want this to be a separate stage or is this fine?

Reply:
2nd question is do we want the pull the stancflags from the Jenkins variable `stanc_flags`

Reply:
I think separate stage with the same conditions as the other stages in this PR, so it doesn't run every time.

Reply:
Done.